### ISSUE TITLE
False positives for "missing" acronym definitions

### ISSUE THREAD

**Issue by jdkato** [jdkato/txtlint#1056]
https://github.com/errata-ai/vale/issues/1002#issuecomment-3602497132

**Comment by theletterf** [comment]
Thanks!

The sample is:

```
either the Hierarchical Navigable Small World ([HNSW](https://arxiv.org/abs/1603.09320)) algorithm
```

My rule is:

```yaml
extends: conditional
message: "'%s' has no definition."
link: https://stunning-adventure-qrvr1k2.pages.github.io/tech-writing-guidelines/grammar-spelling
level: suggestion
ignorecase: false
scope: text
first: '\b([A-Z]{3,5})\b'
second: '(?:\b[A-Za-z][a-z]*(?:-[a-z]+)* )+\(([A-Z]{3,5})\)'
```
